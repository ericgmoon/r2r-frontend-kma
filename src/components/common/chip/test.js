import React from "react";
import renderer from "react-test-renderer";
import sinon from "sinon";
import { shallow } from "enzyme";
import StyledChip from "./styles";
import Chip from "./index";

test("should not capitalise text", () => {
  const node = shallow(<Chip label="cHiP" />);
  expect(node.find(StyledChip).prop("title")).toEqual("cHiP");
});

test("should trigger onPress", () => {
  const callback = sinon.spy();
  const node = shallow(
    <Chip className="chip" onPress={callback} label="chip" />
  );
  expect(node.find(".chip")).toBeDefined();
  node.find(".chip").simulate("press");
  sinon.assert.called(callback);
});

describe("renders correctly", () => {
  test("without props", () => {
    const node = renderer.create(<Chip />).toJSON();
    expect(node).toMatchSnapshot();
  });

  test("with label", () => {
    const node = renderer.create(<Chip label="chip" />).toJSON();
    expect(node).toMatchSnapshot();
  });

  test("when active", () => {
    const node = renderer.create(<Chip label="chip" active />).toJSON();
    expect(node).toMatchSnapshot();
  });

  test("when static", () => {
    const node = renderer.create(<Chip label="chip" isStatic />).toJSON();
    expect(node).toMatchSnapshot();
  });

  test("when active with custom active colour", () => {
    const node = renderer
      .create(<Chip label="chip" active activeColor="#ffdb58" />)
      .toJSON();
    expect(node).toMatchSnapshot();
  });

  test("with custom styles", () => {
    const node = renderer
      .create(
        <Chip label="chip" chipStyle={{ paddingTop: 32, paddingButton: 32 }} />
      )
      .toJSON();
    expect(node).toMatchSnapshot();
  });
});
