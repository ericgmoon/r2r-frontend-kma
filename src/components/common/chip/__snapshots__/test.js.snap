// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when active 1`] = `
<View
  pointerEvents="auto"
>
  <View
    style={
      Object {
        "borderRadius": 36,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#DF6A6A",
            "borderColor": "#2089dc",
            "borderRadius": 36,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 12,
              "fontWeight": "700",
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          chip
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly when active with custom active colour 1`] = `
<View
  pointerEvents="auto"
>
  <View
    style={
      Object {
        "borderRadius": 36,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffdb58",
            "borderColor": "#2089dc",
            "borderRadius": 36,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 12,
              "fontWeight": "700",
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          chip
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly when static 1`] = `
<View
  pointerEvents="none"
>
  <View
    style={
      Object {
        "borderRadius": 36,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ECECEC",
            "borderColor": "#2089dc",
            "borderRadius": 36,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 12,
              "fontWeight": "700",
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          chip
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly with custom styles 1`] = `
<View
  pointerEvents="auto"
>
  <View
    style={
      Object {
        "borderRadius": 36,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ECECEC",
            "borderColor": "#2089dc",
            "borderRadius": 36,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
            "paddingBottom": 8,
            "paddingButton": 32,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 32,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 12,
              "fontWeight": "700",
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          chip
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly with label 1`] = `
<View
  pointerEvents="auto"
>
  <View
    style={
      Object {
        "borderRadius": 36,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ECECEC",
            "borderColor": "#2089dc",
            "borderRadius": 36,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 12,
              "fontWeight": "700",
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          chip
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly without props 1`] = `
<View
  pointerEvents="auto"
>
  <View
    style={
      Object {
        "borderRadius": 36,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ECECEC",
            "borderColor": "#2089dc",
            "borderRadius": 36,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          }
        }
      />
    </View>
  </View>
</View>
`;
